stripe/src/test/java/com/stripe/android/model/PaymentIntentTest.java

```diff
@@ -72,71 +71,174 @@
+            "  },\n" +
+            "  \"source\": \"src_1E884r2eZvKYlo2CTft0qEyY\",\n" +
+            "  \"statement_descriptor\": \"PaymentIntent Statement Descriptor\",\n" +
+            "  \"status\": \"succeeded\",\n" +
+            "  \"transfer_data\": null,\n" +
+            "  \"transfer_group\": null\n" +
+            "}";
+
+    private static final String PARTIAL_PAYMENT_INTENT_WITH_REDIRECT_URL_JSON = "{\n" +
+            "\t\"id\": \"pi_Aabcxyz01aDfoo\",\n" +
+            "\t\"object\": \"payment_intent\",\n" +
+            "\t\"status\": \"requires_action\",\n" +
+            "\t\"next_action\": {\n" +
+            "\t\t\"type\": \"redirect_to_url\",\n" +
+            "\t\t\"redirect_to_url\": {\n" +
+            "\t\t\t\"url\": \"https://example.com/redirect\"\n" +
+            "\t\t}\n" +
+            "\t}\n" +
+            "}";
+
+    private static final String PARTIAL_PAYMENT_INTENT_WITH_AUTHORIZE_WITH_URL_JSON = "{\n" +
+            "\t\"id\": \"pi_Aabcxyz01aDfoo\",\n" +
+            "\t\"object\": \"payment_intent\",\n" +
+            "\t\"status\": \"requires_action\",\n" +
+            "\t\"next_action\": {\n" +
+            "\t\t\"type\": \"authorize_with_url\",\n" +
+            "\t\t\"authorize_with_url\": {\n" +
+            "\t\t\t\"url\": \"https://example.com/redirect\"\n" +
+            "\t\t}\n" +
+            "\t}\n" +
+            "}";
+
+    private static final List<String> ALLOWED_SOURCE_TYPES = new ArrayList<String>() {{
         add("card");
     }};
 
-    private static final Map<String, Object> EXAMPLE_PAYMENT_INTENT_MAP = new HashMap<String, Object>() {{
-        put("id", "pi_1CkiBMLENEVhOs7YMtUehLau");
-        put("object", "payment_intent");
-        put("allowed_source_types", ALLOWED_SOURCE_TYPES);
-        put("amount", 1000L);
-        put("canceled_at", 1530839340L);
-        put("client_secret", "pi_1CkiBMLENEVhOs7YMtUehLau_secret_s4O8SDh7s6spSmHDw1VaYPGZA");
-        put("confirmation_method", "publishable");
-        put("created", 1530838340L);
-        put("currency", "usd");
-        put("description", "Example PaymentIntent charge");
-        put("livemode", false);
-        put("source", "src_1CkiC3LENEVhOs7YMSa4yx4G");
-        put("capture_method", "automatic");
-        put("status", "succeeded");
-    }};
-
-    private PaymentIntent mPaymentIntent;
-
-    @Before
-    public void setup() {
-        mPaymentIntent = PaymentIntent.fromString(EXAMPLE_PAYMENT_INTENT_SOURCE);
-        assertNotNull(mPaymentIntent);
-    }
+    private static final Map<String, Object> PAYMENT_INTENT_WITH_SOURCE_MAP =
+            new HashMap<String, Object>() {{
+                put(PaymentIntent.FIELD_ID, "pi_1CkiBMLENEVhOs7YMtUehLau");
+                put(PaymentIntent.FIELD_OBJECT, "payment_intent");
+                put(PaymentIntent.FIELD_ALLOWED_SOURCE_TYPES, ALLOWED_SOURCE_TYPES);
+                put(PaymentIntent.FIELD_PAYMENT_METHOD_TYPES, new ArrayList<>());
```

Comment:
i think this is just for testing but the values of allowed_source_types and payment_method_types are always mirrored